### PR TITLE
Remove dbase new

### PR DESCRIPTION
**Pull request by MiguelAHM**
Remove unused dbase_new from split files:

* Swapped "dir.rpsl.export.external.legacy" by "dir.rpsl.export.external" and setting to dbase. We need to do the same change in db-operational. We have to set the export.external to dbase (instead dbase_new) and remove the legacy.

### CODE REVIEW

**Comment by MiguelAHM** [comment]
Test are ok
https://gitlab.ripe.net/swe-database-team/whois/-/pipelines/382567

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java [hunk 1]
```diff
@@ -1,29 +1,99 @@
...
+
         for (int i = 0; i < attributes.size(); i++) {
             RpslAttribute replacement = attributes.get(i);
 
-            replacement = dummifyOrgName(rpslObject, replacement);
-            replacement = dummifyDescr(replacement);
-            replacement = dummifyRemarks(replacement);
-            replacement = dummifyOwner(replacement);
+            replacement = dummifyAttributes(rpslObject, attributes, i, replacement);
+
```
Comment (maggarwal13) [inline whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java:58]: What does i signifies? Can you not pass the counter of for loop in a method. It is not so readable

Comment (maggarwal13) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java:58]: The original code seems more readable for me

File: whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/task/export/ExportFileWriterFactory.java [hunk 2]
```diff
@@ -27,43 +26,33 @@ class ExportFileWriterFactory {
...
-    private final String externalExportDir;
     private final String internalExportDir;
 
     @Autowired
-    ExportFileWriterFactory(final DummifierNrtm dummifierNrtm, final DummifierCurrent dummifierCurrent,
+    ExportFileWriterFactory(final DummifierNrtm dummifierNrtm,
                             @Value("${dir.rpsl.export.internal}") final String internalExportDir,
                             @Value("${dir.rpsl.export.external}") final String externalExportDir,
-                            @Value("${dir.rpsl.export.external.legacy}") final String legacyExternalExportDir,
                             @Value("${whois.source}") final String source,
```
Comment (maggarwal13) [inline whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/task/export/ExportFileWriterFactory.java:38]: If you are renaming the property, classname should also be renamed ? DummifyLegacy to DummifyCurrent or something? I would prefer DummifySplitFiles if this dummifier is only used by splitfiles?

File: whois-scheduler/src/test/java/net/ripe/db/whois/scheduler/task/export/ExportFileWriterFactoryTest.java [hunk 3]
```diff
@@ -57,12 +55,11 @@ public void createExportFileWriters() {
...
         assertThat(files, not(nullValue()));
-        assertThat(files.length, is(3));
+        assertThat(files.length, is(2));
 
         for (final File file : files) {
             if (! (file.getAbsolutePath().endsWith("internal")
-                    || file.getAbsolutePath().endsWith("dbase")
-                    || file.getAbsolutePath().endsWith("dbase_new"))) {
+                    || file.getAbsolutePath().endsWith("dbase"))) {
                 fail("Unexpected folder: " + file.getAbsolutePath());
```
Comment (maggarwal13) [inline whois-scheduler/src/test/java/net/ripe/db/whois/scheduler/task/export/ExportFileWriterFactoryTest.java:63]: can you refactor the code to assertThat that it is just dbase and internal in any order?

Comment (MiguelAHM) [reply whois-scheduler/src/test/java/net/ripe/db/whois/scheduler/task/export/ExportFileWriterFactoryTest.java]: done

File: whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/task/export/ExportFileWriterFactory.java (hunk 2 above)
Comment (MiguelAHM) [reply whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/task/export/ExportFileWriterFactory.java:38]: Done

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java (hunk 1 above)
Comment (MiguelAHM) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java:63]: Done

File: whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java [hunk 4]
```diff
@@ -1,17 +1,56 @@
...
+import static net.ripe.db.whois.common.rpsl.AttributeType.MNT_NFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.NOTIFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.PERSON;
+import static net.ripe.db.whois.common.rpsl.AttributeType.PHONE;
+import static net.ripe.db.whois.common.rpsl.AttributeType.REF_NFY;
+import static net.ripe.db.whois.common.rpsl.AttributeType.UPD_TO;
 
 @Component
-public class DummifierRC extends DummifierCurrent {
+public class DummifierRC implements Dummifier {
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java:30]: At the risk of *more* confusion, could you rename to `DummifierCurrent` ? *This* is now the current (default) dummifier.

Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/DummifierRC.java:30]: To answer my own question: in fact `DummifierNrtm` is used to create the split files. Leave this class name as-is.

**Review by eshryane — APPROVED** [review]

**Review by maggarwal13 — APPROVED** [review]

**Pull request merged**